src/otx/core/exporter/native.py

```diff
@@ -46,14 +46,16 @@ def to_openvino(
         output_dir: Path,
         base_model_name: str = "exported_model",
         precision: OTXPrecisionType = OTXPrecisionType.FP32,
+        export_args: dict[str, Any] | None = None,
     ) -> Path:
         """Export to OpenVINO Intermediate Representation format.
 
         In this implementation the export is done only via standard OV/ONNX tools.
         """
-        dummy_tensor = torch.rand(self.input_size).to(next(model.parameters()).device)
-
         if self.via_onnx:
+            if export_args is None:
+                export_args = {"args": torch.rand(self.input_size).to(next(model.parameters()).device)}
```

Comment:
Why is this needed?

Reply:
I don't like this "export_args " approach everywhere. Let's consider to either remove it or change to "sample_input" if input needs to be modified. But, I can't understand in which cases we should modify dummy input for export?

Reply:
For visual prompting, there are two types of inputs, one is an normal image for image encoder like other tasks and the other is image embeddings from image encoder and 4 args of prompt inputs for decoder.
https://github.com/openvinotoolkit/training_extensions/blob/2708d7ba16465e35cd7665da3043223ecc2dde3f/src/otx/core/model/entity/visual_prompting.py#L85-L104

Reply:
@kprokofi For visual prompting, there are two types of inputs, one is a normal image for image encoder like other tasks and the other is image embeddings from image encoder and 4 args of prompt inputs for decoder.
https://github.com/openvinotoolkit/training_extensions/blob/2708d7ba16465e35cd7665da3043223ecc2dde3f/src/otx/core/model/entity/visual_prompting.py#L85-L104

Reply:
Can you define a new property, `VisualPromptModel._export_parameters` add a new `Exporter` for `VisualPromptModel._exporter` to solve your case (including additional input tensors not image)?

Reply:
As far as I understood, `_export_parameters` is for building `Exporter`, not `_exporter.export`.
https://github.com/openvinotoolkit/training_extensions/blob/2708d7ba16465e35cd7665da3043223ecc2dde3f/src/otx/core/model/entity/visual_prompting.py#L120-L156
So, I added `export_args` (current `example_inputs`) for export execution.
Could you suggest a way what you said?

Reply:
```python
class VisualPromptExporter(OTXExporter):
     def __init__(self, dummy_decoder_input, ...):
         ...

...

class VisualPromptModel:
    @property
    def _export_parameters(self):
         return {"dummy_decoder_input": ..., ...}
    
    @property
    def _exporter(self):
         return VisualPromptExporter(**self._export_parameters)
    
```

Reply:
Is it okay to add new exporter for this task?
It seems that exporters are distinguished with frameworks, mmdeploy and torch.

Reply:
I cannot see any reason not to add.

Reply:
@harimkang @jaegukhyun @sungmanc  What do you think about adding a new exporter only for visual prompting?
As @vinnamkim said, it definitely makes setting parameters clearer, but I need more opinions because exporters seems to be distinguished with frameworks, not tasks.

Reply:
In the very first place, there are no such restriction that the exporters should be distinguished by framework. The reason why we make the exporters is to lower code complexity of `OTXModel`. However, in your case, serving multiple dummy input tensors during exportation is not covered by the existing exporters. To resolve this, it seems better to create a new exporter dedicated to cover your use-case, rather than overhauling a bunch of code in multiple places to extend the functionality of the existing one.

Reply:
I agree with your opinion, but I'm just curious about others' thoughts.
For now, I'll add a new exporter for this case, and let's discuss how to handle different types of exporters and decide on policies later.

Reply:
To add my opinion, I think this is about who should be in charge of preparing export argument. When we implemented exporter classes, it's true to distinguish export class based on framework (torch + OV and mmdeploy + OV). Of course, there is no restriction that exporter should be distinguished based on framework, but it seems a little weird that det and i-seg model uses mmdeploy exporter to use mmdeploy but VP uses separated exporter even if it uses torch + OV like other tasks. Anyway, back to topic about responsibility, I think that preparing export argument by VP task model isn't wrong because it's nature to prepare necessary things for own purpose. I think making a new class for VP export also can be other way to deal with this but, as I said, it's a little awkward if we look all exports class implementation including VP exporter in my perspective.

Reply:
The problem is that we already prepared this mechanism to re-define these two properties in the derived class:
https://github.com/openvinotoolkit/training_extensions/blob/98fb7126085ae5766ad1bd827c555ec9cc698c29/src/otx/core/model/entity/base.py#L266-L300
It must have all the degree of freedom to adapt every model exportation use-case with this mechanism. However, this PR is ignoring this mechanism from the front and is trying to dig out a new interface by expanding `OTXModel.export()` and `OTXExporter.export()`.

Reply:
> @harimkang @jaegukhyun @sungmanc What do you think about adding a new exporter only for visual prompting? As @vinnamkim said, it definitely makes setting parameters clearer, but I need more opinions because exporters seems to be distinguished with frameworks, not tasks.

I prefer to use the additional class for the VisualPrompting task since this task includes special cases.

Reply:
> It must have all the degree of freedom to adapt every model exportation use-case with this mechanism. However, this PR is ignoring this mechanism from the front and is trying to dig out a new interface by expanding OTXModel.export() and OTXExporter.export().

Yeap. In my thought, to make design more clean, we need to change overall code but it should be considered in other PR. I don't have any objection if other team member prefer new class design.